Fix remaining CPU operators for non-standard bools

peterbell10 opened the pull request:
Stack from [ghstack](https://github.com/ezyang/ghstack) (oldest at bottom):
* #79393
* #79392
* #79391
* __->__ #79390

In addition to using `c10::load` in more kernels, this specializes
`Vectorized<bool>::loadu` to first load as `int8_t` then convert to
bool, in the same way as `c10::load` does.

For `unique_cpu`, the values are loaded inside of the
`std::unordered_set` constuctor, so I've added a separate function
specifically for bools. This fixes the loading issue and is faster
because it uses the fact that bools can only have one of two values.

peterbell10 committed:
Fix remaining CPU operators for non-standard bools

In addition to using `c10::load` in more kernels, this specializes
`Vectorized<bool>::loadu` to first load as `int8_t` then convert to
bool, in the same way as `c10::load` does.

[ghstack-poisoned]

peterbell10 committed:
Skip tracing tests for equal on "Fix remaining CPU operators for non-standard bools"


In addition to using `c10::load` in more kernels, this specializes
`Vectorized<bool>::loadu` to first load as `int8_t` then convert to
bool, in the same way as `c10::load` does.

[ghstack-poisoned]

peterbell10 committed:
Update on "Fix remaining CPU operators for non-standard bools"


In addition to using `c10::load` in more kernels, this specializes
`Vectorized<bool>::loadu` to first load as `int8_t` then convert to
bool, in the same way as `c10::load` does.

[ghstack-poisoned]

peterbell10 committed:
Skip meta tests for torch.equal on "Fix remaining CPU operators for non-standard bools"


In addition to using `c10::load` in more kernels, this specializes
`Vectorized<bool>::loadu` to first load as `int8_t` then convert to
bool, in the same way as `c10::load` does.

[ghstack-poisoned]

peterbell10 committed:
Update on "Fix remaining CPU operators for non-standard bools"


In addition to using `c10::load` in more kernels, this specializes
`Vectorized<bool>::loadu` to first load as `int8_t` then convert to
bool, in the same way as `c10::load` does.

[ghstack-poisoned]

peterbell10 committed:
Update on "Fix remaining CPU operators for non-standard bools"


In addition to using `c10::load` in more kernels, this specializes
`Vectorized<bool>::loadu` to first load as `int8_t` then convert to
bool, in the same way as `c10::load` does.

[ghstack-poisoned]

peterbell10 committed:
Update on "Fix remaining CPU operators for non-standard bools"


In addition to using `c10::load` in more kernels, this specializes
`Vectorized<bool>::loadu` to first load as `int8_t` then convert to
bool, in the same way as `c10::load` does.

[ghstack-poisoned]

peterbell10 committed:
Rebase on "Fix remaining CPU operators for non-standard bools"


In addition to using `c10::load` in more kernels, this specializes
`Vectorized<bool>::loadu` to first load as `int8_t` then convert to
bool, in the same way as `c10::load` does.

[ghstack-poisoned]

mruberry reviewed (approved):
Stamped! Thanks @peterbell10!

peterbell10 committed:
Update on "Fix remaining CPU operators for non-standard bools"


In addition to using `c10::load` in more kernels, this specializes
`Vectorized<bool>::loadu` to first load as `int8_t` then convert to
bool, in the same way as `c10::load` does.

[ghstack-poisoned]

peterbell10 committed:
Update on "Fix remaining CPU operators for non-standard bools"


In addition to using `c10::load` in more kernels, this specializes
`Vectorized<bool>::loadu` to first load as `int8_t` then convert to
bool, in the same way as `c10::load` does.

For `unique_cpu`, the values are loaded inside of the
`std::unordered_set` constuctor, so I've added a separate function
specifically for bools. This fixes the loading issue and is faster
because it uses the fact that bools can only have one of two values.

[ghstack-poisoned]

peterbell10 committed:
Rebase on "Fix remaining CPU operators for non-standard bools"


In addition to using `c10::load` in more kernels, this specializes
`Vectorized<bool>::loadu` to first load as `int8_t` then convert to
bool, in the same way as `c10::load` does.

For `unique_cpu`, the values are loaded inside of the
`std::unordered_set` constuctor, so I've added a separate function
specifically for bools. This fixes the loading issue and is faster
because it uses the fact that bools can only have one of two values.

[ghstack-poisoned]
